### PR TITLE
Dockerize app

### PR DESCRIPTION
**Pull request by ViralOne**
## Description
Create Dockerfile and docker-compose file to deploy app easily.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ && apk add \
+    build-base \
+    tzdata \
+    libxml2-dev \
+    libxslt-dev \
+    postgresql-client \
+    postgresql-dev
+RUN bundle install
+ONBUILD rails db:setup
+CMD ["rails", "server", "-b", "0.0.0.0"]
```
Comment (ViralOne) [inline Dockerfile]: Remove last line

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    container_name: freakscatalog-db
+    image: postgres:13.1-alpine
+    volumes:
+      - db-data:/var/lib/postgresql/data
+  api:
+    build: .
+    command: rails server -b 0.0.0.0
+    container_name: freakscatalog-api
+    env_file:
+       - .env.dev
```
Comment (ViralOne) [inline docker-compose.yml]: Use .env file instead of .env.dev

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    environment:
+            DATABASE_URL: ${DATABASE_URL}
+    ports:
+      - "3000:3000"
+    volumes:
+      - ./:/home/app
```
Comment (ViralOne) [inline docker-compose.yml:20]: Check if /home exists

**Comment by cazacutudor** [comment]
Way better than the first iteration :1234:,

There's still a slight issue, that we have to fix, before merging this pull-request, please see the image attached below,
image
cc @ViralOne

**Comment by ViralOne** [comment]
> Way better than the first iteration 1234,
> 
> There's still a slight issue, that we have to fix, before merging this pull-request, please see the image attached below, image cc @ViralOne

I think this is because you don't have the .env file that I created. I had to create a slightly different DATABASE_URL from what was in the file, otherwise it couldn't connect.

**Comment by cazacutudor** [comment]
> > Way better than the first iteration 1234,
> > There's still a slight issue, that we have to fix, before merging this pull-request, please see the image attached below, image cc @ViralOne
> 
> I think this is because you don't have the .env file that I created. I had to create a slightly different DATABASE_URL from what was in the file, otherwise it couldn't connect.

Aha! So that's the secret, I'm wondering now, how the rest of the team will be able to run it, without having the magic `.env`? :grinning: 

Also, might be helpful for our colleges to have some guidelines, on how to start the app using the new docker setup. I'm sure they will be very grateful for that. This article [about readmes](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-readmes), might help you to kick things off.

File: .gitignore [hunk 4]
```diff
@@ -7,6 +7,8 @@
 .env
 .env.test
 
+/vendor
+
```
Comment (cazacutudor) [inline .gitignore]: What is this `vendor` directory and why do we add it to .gitignore ?

File: README.md [hunk 5]
```diff
@@ -50,4 +50,41 @@ Otherwise it is enough to place the next order
...
+```
+
+### Step 3. Install PostgreSQL Helm Chart
+```
+$ helm install freakscatalog-api-db bitnami/postgresql --set persistence.existingClaim=postgresql-pv-claim --set volumePermissions.enabled=true
+```
+
+### Step 4. Install project Helm Chart
+```
+$ helm install freakscatalog-api helm/freakscatalog-api/
```
Comment (cazacutudor) [inline README.md]: This block seems to be out of the PR scope. Let's move it, please.

**Review by cazacutudor — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (ViralOne) [reply .gitignore]: While working, `/vendor` folder was showing +7000 new files and those were mainly only gems for ruby, but this was my fault while I was testing it generated that specific folder. I will remove that in the next commit.

**Comment by cazacutudor** [comment]
The database setup still not working, 

image


I've patched your PR, please fetch the latest commit and have a look. cc @ViralOne

**Comment by ViralOne** [comment]
I fetched the last commit and tested on my local machine, everything it's working properly.

**Pull request merged**